.github/workflows/dispatch.yml

```diff
@@ -30,3 +30,10 @@ jobs:
     if: ${{ github.ref_protected == true }}
     uses: ./.github/workflows/release.yml
     secrets: inherit
+  
+  call-release-api:
+    name: "Release API"
+    needs: call-release
+    uses: ./.github/workflows/release-api.yml
```

Comment:
This one is not compatible with `on.push` as you are using below in `release-api.yml`. Please, pick only one of calling method. I assume, it should be independent from releasing process and reflects only changes in `api/` directory.